lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1727,6 +1727,22 @@ static void optimizeSliceOfSplat(Function *F) {
   }
 }
 
+/// Replace SliceNode when the input is Constant with another constant node
+/// Slice(constant1) -> constant2
+static void optimizeSliceOfConstant(Function *F) {
+  for (auto &node : F->getNodes()) {
+    auto *sliceNode = dyn_cast<SliceNode>(&node);
+    if (!sliceNode)
+      continue;
+    auto *constantNode = dyn_cast<Constant>(sliceNode->getInput());
+    if (!constantNode)
+      continue;
+    auto *newConstantNode = F->getParent()->createConstant(
```

Comment:
What happens with the payload of this new constant node? It needs to be backed by a tensor that represents the slice of the original constant, or?

But I don't see here how you initialize this payload.

Reply:
I missed that. Thank you for pointing that out. I will make the necessary changes and update.

Reply:
I have spent the last couple of days reading existing code and through the documentation. 
I have been however stuck at the point of initializing the payload of the newConstantNode to be the sliced version of the original constant.
Could you give me pointers regarding finding the sliced payload of a tensor?

Reply:
@SumedhArani `Constant` has a `getPayload` method returning a tensor with a payload of this constant. Now you need to extract a proper slice from it. And `Tensor` has a number of methods for manipulating its content, e.g. `copySlice`, `copyConsecutiveSlices`, etc. You should try to use those method to form a new Tensor representing the sliced version of the original constant. Once you have, you can use `createConstant` that takes a tensor as one of the parameters or you can create a `Constant` using type, shape, etc and then set its payload by means of calling the `Constant::assign` method.

Reply:
Thanks @opti-mix !

It's of much help. Thanks for taking the time to respond back to me! 👍